### PR TITLE
Quote the "group" field in the UserCreateCommand statement

### LINKED ISSUE #2849: UserCreateCommand SQL error
**Affected version(s)**

Contao 4.11

**Description**

The UserCreateCommand has an issue with reserved keywords:

```
CRITICAL  [console] Error thrown while running command "contao:user:create". Message: "An exception occurred while executing 'INSERT INTO tl_user (tstamp, name, email, username, password, language, backendTheme, admin, pwChange, dateAdded, groups) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [1615137394, "Name", "email", "username", "$2y$13.", "en", "flexible", true, false, 1615137394, ""]:

SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'groups) 
```

This is because ["groups" is a (new) reserved keyword](https://dev.mysql.com/doc/refman/8.0/en/keywords.html).

However, we are already using DBAL, from which I thought it is taking care of escaping?

https://github.com/contao/contao/blob/6cdb7fce22c393ba19e0ca3b9237398b9bbba050/core-bundle/src/Command/UserCreateCommand.php#L280-L293

### PR DESCRIPTION
**Pull request by richardhj**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #2849
| Docs PR or issue | 

Fixes a SQL error caused by missing parameter quoting.

The source code of DBAL demonstrates that DBAL is not performing any quoting in this case: https://github.com/doctrine/dbal/blob/7da29b1c5b6ec4fe8636c0a022d98d68058a9053/src/Connection.php#L732

Also see doctrine/dbal#4357 (don't know how relevant this issue is)

### CODE REVIEW

**Review by bytehead — DISMISSED** [review]

**Review by fritzmg — DISMISSED** [review]

**Comment by m-vo** [comment]
This is so weird - is this at least fixed in `doctrine/dbal` 3.0? I get that they have issues with BC but having to quote identifiers is exactly the thing an abstraction layer must handle. And, apparently, that's already the case in the schema tool (see https://github.com/contao/contao/issues/1696#issuecomment-623492105).

**Comment by richardhj** [comment]
> This is so weird - is this at least fixed in `doctrine/dbal` 3.0? I get that they have issues with BC but having to quote identifiers is exactly the thing an abstraction layer must handle. And, apparently, that's already the case in the schema tool (see [#1696 (comment)](https://github.com/contao/contao/issues/1696#issuecomment-623492105)).

Maybe this is different for queries that are executed from a QueryBuilder query. Idk.

**Comment by ausi** [comment]
> This is so weird - is this at least fixed in `doctrine/dbal` 3.0?

No. See https://github.com/doctrine/dbal/blob/3.0.0/src/Connection.php#L690

File: core-bundle/src/Command/UserCreateCommand.php [hunk 1]
```diff
@@ -277,21 +277,28 @@ private function persistUser(string $username, string $name, string $email, stri
...
+            'password' => $hash,
+            'language' => $language,
+            'backendTheme' => 'flexible',
+            'admin' => $isAdmin,
+            'pwChange' => $pwChange,
+            'dateAdded' => $time,
+        ];
+
+        if (!$isAdmin && !empty($groups)) {
+            $data[$this->quote('groups')] = serialize(array_map('strval', $groups));
```
Comment (leofeyer) [inline core-bundle/src/Command/UserCreateCommand.php]:
```suggestion
            $data[$this->connection->quoteIdentifier('groups')] = serialize(array_map('strval', $groups));
```

No need to add a private method in this case.

**Review by leofeyer — CHANGES_REQUESTED** [review]

**Review by m-vo — APPROVED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @richardhj.

**Pull request merged**